### DISCUSSION TITLE
NG Alert allow configuring variable parameters for Dashboard and Graph Panel metadata

### DISCUSSION THREAD

**Discussion by d-shi** [grafana/grafana#42099] [Alerting]
We are using NG Alert's multi dimensional alerting to simplify our alert rules across multiple environments and clusters, which is great! We would like to associate these alerts to respective dashboards and graph panels, currently doing so with the dashboard UID and panel UID metadata in the step 4 of the alert definition. The associated dashboards and panels have template variables across the same dimensions as in the alert definition.

However, since these fiels only accept UID, there is no way to specify which dashboard variables to use when generating the dashboard and panel links. For example, if we have a dashboard with cluster as a template variable which shows cluster A by default and an associated alert across the cluster dimension, then if the alert fires for cluster B, the dashboard and panel URL generated in the alert body still bring up the dashboard and panel displaying cluster A. This can become very confusing, especially in the heat of triaging a production issue.

It would be great to be able to use the same alert template variables (https://grafana.com/docs/grafana/latest/alerting/unified-alerting/alerting-rules/alert-annotation-label/) in the dashboard and panel alert metadata, so that the relevant dashboard and panel URLs can be generated depending on which time series triggered the alert.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
